### DISCUSSION TITLE
Using RenderSection in a module for MVC views

### DISCUSSION THREAD

**Discussion by LRathje** [OrchardCMS/OrchardCore#8927] [Q&A] [answered]
Hi

I have set up a well working enviroment with a main project with OrchardCoreCMS and reference a MVC module from where I can get content from the CMS via ContentManagement to mix with my MVC content and models. It's actually working quite well as a decoupled CMS because I'm not that much into Liquid and pages yet. So I prefer currently the MVC and Views.

But I have come accross a problem, which I hope someone can help with. It's when you start using views with a _Layout.cshtml, a middelview and a contentview. I want one middleview to show a menu for some contentpages, but another middleview to show another menu and a third middleview to show no menu. I will like to here use `@await RenderSectionAsync("Menu", required: false)` in the layout and `@section Menu {<somemenu>}` in the middleview.

Then I get this error:
The following sections have been defined but have not been rendered by the page at` '/Areas/...../Views/Shared/_Layout.cshtml': 'Menu'. To ignore an unrendered section call IgnoreSection("sectionName").`

Please observe this is not for page, but for mvc views, and I'm having this in my _ViewImports.cshtml: `@inherits OrchardCore.DisplayManagement.Razor.RazorPage<TModel>`

I hope this is just me and very simple to solve.

Thanks Lars

**Comment by jtkech** [comment, marked as answer]
There some related issues e.g. #3534

In our custom `Razor.RazorPage` we don't replace the `DefineSection()` base method so that the `@section` directive could be supported, at some point it was done through #5413 but has been reverted, as i remember because it is a non async method and we had no choice to use `.GetAwaiter().GetResult()`

You can use our zone tag helper instead e.g. https://github.com/OrchardCMS/OrchardCore/issues/2673#issuecomment-446704616

**Reply by LRathje** [reply]
Thanks for a quick response @jtkech. Can you use the zone tag in pure mvc views instead of `@section` and still call it with `RenderSectionAsync`?

**Reply by LRathje** [reply]
Ahh - just tested it and it works flawlessly!!
Then I feel I already have taken a small first step into the Liquid world ;-).

Thanks a lot @jtkech

**Reply by jtkech** [reply]
No problem ;) Then mark it as answered if it's okay for you
